File: Core/Entities/Campground.cs
using Core.Interfaces;

namespace Core.Entities;

public class Campground : BaseEntity, IDtoConvertible
{
    public required string Name { get; set; }
    public required string Description { get; set; }
    public required string PictureUrl { get; set; }
    public ICollection<CampgroundAmenity> Amenities { get; set; } = [];
    public ICollection<Campsite> Campsites { get; set; } = [];
}

File: client/src/app/shared/models/campground.ts
import {Campsite} from './campsite';
import {CampgroundAmenity} from './campgroundAmenity';

export type Campground = {
  id: number;
  name: string;
  description: string;
  pictureUrl: string;
  amenities: CampgroundAmenity[];
  campsites: Campsite[];
}
